### PR TITLE
Drop "experimental" from `README.md`

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/conda/conda-libmamba-solver/issues/188

### Description

This is no longer experimental. So drop this caveat from the README.

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ The fast mamba solver, now in conda!
 
 ## What is this exactly?
 
-conda-libmamba-solver is a new (experimental) solver for the
+conda-libmamba-solver is a new solver for the
```
Comment (jakirkham) [inline README.md:12]: Dropped "experimental", but left "new". Idk if we want to drop both. Please let me know what you think :)

**Comment by jakirkham** [comment]
Looks like the other CI runs need approval

Also not sure what this [`pre-commit.ci` error]( https://results.pre-commit.ci/run/github/449789872/1681757187.oCjr35FcQky-obfjltlPKg ) means:

```
Insert license in comments...............................................Failed
- hook id: insert-license
- exit code: 1
- files were modified by this hook
```

File: README.md (hunk 1 above)
Comment (jezdez) [reply README.md:12]: Nah, this is fine, thank you!

**Review by jezdez — APPROVED** [review]

**Comment by jaimergp** [comment]
> Looks like the other CI runs need approval
> 
> Also not sure what this [`pre-commit.ci` error](https://results.pre-commit.ci/run/github/449789872/1681757187.oCjr35FcQky-obfjltlPKg) means:
> 
> ```
> Insert license in comments...............................................Failed
> - hook id: insert-license
> - exit code: 1
> - files were modified by this hook
> ```

This is being handled in #187, no worries.

**Pull request merged**